docs: flesh out landing page and add tutorial (#61)
canonical/jubilant: docs/tutorial/getting-started.md

# Getting started with Jubilant

> In this tutorial, we will learn how to install Jubilant, run Juju commands with it, and write a simple charm integration test.

Reviewer: Minor readability suggestions:

```suggestion
In this tutorial, we'll learn how to install Jubilant, use it to run Juju commands, and write a simple charm integration test.
```

The tutorial assumes that you have a basic understanding of Juju and have already installed it. [Learn how to install the Juju CLI.](https://canonical-juju.readthedocs-hosted.com/en/latest/user/howto/manage-juju/#install-juju)

## Install Jubilant

> Jubilant is published to PyPI, so you can install and use it with your favourite Python package manager:

Reviewer: "favourite" - shall we start using US English throughout Jubilant?

Author: Yeah, given that's the Company Docs Mandate now, I guess so. Done.

```
$ pip install jubilant
# or
$ uv add jubilant
```

> ## Ensure it's working
> 
> To ensure Jubilant is working, use it to add a Juju model and check its status:

Reviewer: Minor - To me, "ensure" suggests that I am responsible for making sure that Jubilant is working. How about:

```suggestion
## Check your setup

To check that Jubilant is working, use it to add a Juju model and check its status:
```

Here is a model-setup fixture called `juju`, which you would normally define in [`conftest.py`](https://docs.pytest.org/en/stable/reference/fixtures.html#conftest-py-sharing-fixtures-across-multiple-files):

```python
@pytest.fixture
def juju():
    with jubilant.temp_model() as juju:
        yield juju
```

An integration test would use the fixture, operating on the temporary model:

```python
def test_deploy(juju: jubilant.Juju):
    juju.deploy('snappass-test')
    status = juju.wait(jubilant.all_active)
    assert status.apps['snappass-test'].scale == 1
```

You may want to make your fixture [module-scoped](https://docs.pytest.org/en/stable/how-to/fixtures.html#scope-sharing-fixtures-across-classes-modules-packages-or-session), to allow all the tests in one file to share the same temporary model:

```python
@pytest.fixture(scope='module')
def juju():
    ...
```

## Use a custom `wait` condition

> When waiting on a condition with [`Juju.wait`](jubilant.Juju.wait), you can use pre-defined helpers like [](jubilant.all_active) or [](jubilant.any_error), but you can also define a custom condition using the *ready* or *error* parameters. This is typically done with inline `lambda` functions.

Author: Thanks, done. I slightly prefer the *lightness* of emphasis for these, as `monospace` always looks so heavy, so keeping the parameter names as is for now.
